bindings/matrix-sdk-crypto-js/src/responses.rs

```diff
@@ -110,11 +128,27 @@ impl TryFrom<(RequestType, http::Response<Vec<u8>>)> for OwnedResponse {
             RequestType::KeysBackup => {
                 KeysBackupResponse::try_from_http_response(response).map(Into::into)
             }
+
+            RequestType::SigningKeysUpload => {
+                // SigningKeysUploadResponse::try_from_http_response returns a
+                // FromHttpResponseError<UiaaResponse> instead of a
+                // ruma_client_api::error::Error, so we have to hande it
+                // separately. In practice, the application is supposed to
+                // have dealt with UIA before we get here, so we just map it straight into a
+                // regular JsError anyway.
+                return SigningKeysUploadResponse::try_from_http_response(response)
+                    .map(Into::into)
+                    .map_err(JsError::from);
```

Comment:
```suggestion
                SigningKeysUploadResponse::try_from_http_response(response)
                    .map(Into::into)
                    .map_err(JsError::from)
```

Reply:
no, as the comment is trying to say: we need to `return` this directly, otherwise we gen an error about mismatched `match` arms.